edb/edgeql-parser/src/parser/custom_errors.rs

```diff
@@ -0,0 +1,420 @@
+use crate::tokenizer::Kind;
+use crate::{keywords::Keyword, position::Span};
+
+use super::{CSTNode, Context, Error, Parser, StackNode, Terminal};
+
+impl<'s> Parser<'s> {
+    pub(super) fn custom_error(&self, ctx: &Context, token: &Terminal) -> Option<Error> {
+        // self.print_stack(ctx);
+        let ltok = self.get_from_top(0).unwrap();
+
+        if let Some(value) = self.custom_error_from_rule(token, ctx) {
+            return Some(value);
+        }
+
+        if matches!(token.kind, Kind::Keyword(Keyword("explain"))) {
+            let _hint = "Use `analyze` to show query performance details";
+            return Some(Error::new(format!(
+                "Unexpected keyword '{}'",
+                token.text.to_uppercase()
+            )));
+        }
+
+        if let Kind::Keyword(kw) = token.kind {
+            if kw.is_reserved() && !Cond::Production("Expr").check(ltok, ctx) {
+                // Another token followed by a reserved keyword:
+                // likely an attempt to use keyword as identifier
+                return Some(unexpected_reserved_keyword(&token.text, token.span));
+            }
+        };
+
+        // if let CSTNode::Terminal(Terminal {
+        //     kind: Kind::Keyword(kw),
+        //     text,
+        //     span,
+        //     ..
+        // }) = ltok.value
+        // {
+        //     if kw.is_reserved() {
+        //         // Unexpected reserved keyword:
+        //         // likely an attempt to use keyword as identifier
+        //         return Some(unexpected_reserved_keyword(text, *span));
+        //     }
+        // }
```

Comment:
?

Reply:
These are rules that I've ported from the previous implementation, but are already implemented by the new "Missing x" errors.

I guess we will not need them anymore so they can be removed.